site/docs/status.md

```diff
@@ -0,0 +1,362 @@
+| binary         | Y    | Y         | Y    | Y  |
+
+## Data File Formats
+
+| Format  | Java | PyIceberg | Rust | Go |
+|---------|------|-----------|------|----|
+| Parquet | Y    | Y         | Y    | Y  |
+| ORC     | Y    | N         | N    | N  |
+| Puffin  | Y    | N         | N    | N  |
+
+## File IO
+
+| Storage              | Java | PyIceberg | Rust | Go |
+|----------------------|------|-----------|------|----|
+| Local Filesystem     | Y    | Y         | Y    | Y  |
+| Hadoop Filesystem    | Y    | Y         | Y    | Y  |
+| Aws S3               | Y    | Y         | Y    | Y  |
+| Google Cloud Storage | Y    | Y         | Y    | Y  |
+| Memory Fs            | Y    | Y         | Y    | Y  |
+
+## Table Update Operations
+
+### Table Spec V1
+
+| Operation                   | Java | PyIceberg | Rust | Go |
+|-----------------------------|------|-----------|------|----|
+| Update schema               | Y    | N         | Y    | N  |
+| Update partition spec       | Y    | N         | Y    | N  |
+| Update table properties     | Y    | Y         | Y    | N  |
+| Replace sort order          | Y    | N         | N    | N  |
+| Update table location       | Y    | N         | N    | N  |
+| Append data files           | Y    | Y         | N    | N  |
+| Rewrite files               | Y    | Y         | N    | N  |
+| Rewrite manifests           | Y    | Y         | N    | N  |
+| Overwrite files             | Y    | Y         | N    | N  |
+| Row delta                   | Y    | N         | N    | N  |
+| Delete files                | Y    | N         | N    | N  |
+| Update statistics           | Y    | N         | N    | N  |
+| Update partition statistics | Y    | N         | N    | N  |
+| Expire snapshots            | Y    | N         | N    | N  |
+| Manage snapshots            | Y    | N         | N    | N  |
+
+### Table Spec V2
+
+| Operation                   | Java | PyIceberg | Rust | Go |
+|-----------------------------|------|-----------|------|----|
+| Update schema               | Y    | Y         | N    | N  |
+| Update partition spec       | Y    | Y         | N    | N  |
+| Update table properties     | Y    | Y         | Y    | N  |
+| Replace sort order          | Y    | N         | N    | N  |
+| Update table location       | Y    | N         | N    | N  |
+| Append data files           | Y    | Y         | N    | N  |
+| Rewrite files               | Y    | Y         | N    | N  |
+| Rewrite manifests           | Y    | Y         | N    | N  |
+| Overwrite files             | Y    | Y         | N    | N  |
+| Row delta                   | Y    | N         | N    | N  |
+| Delete files                | Y    | Y         | N    | N  |
+| Update statistics           | Y    | N         | N    | N  |
+| Update partition statistics | Y    | N         | N    | N  |
+| Expire snapshots            | Y    | N         | N    | N  |
+| Manage snapshots            | Y    | N         | N    | N  |
+
```

Comment:
With PyIceberg, I've seen many users struggling to figure out why they are running into commit failures because reliability/concurrency safe features have yet to be implemented in the subproject. 

Hence, I think it would help the community tremendously if we could also host information regarding the implementation of these features on the status page.

These are the [reliability features listed in this section](https://iceberg.apache.org/docs/1.5.2/reliability/#reliability) like isolation levels, and commit retries/concurrent writes support